Cooper/Controllers/RegistrationController.cs

```diff
@@ -26,7 +25,12 @@ public RegistrationController()
         [ProducesResponseType(201)]
         public IActionResult Post([FromBody]User user)
         {
-            if (!ModelState.IsValid || user.Id != 0)
+            bool nicknameExists = userRepository.IfNicknameExists(user.Nickname);       // validation that we don't create user with the same nickname
+            bool emailExists = userRepository.IfEmailExists(user.Email);
+
+            // TODO: divide this statement into three and send the proper explanation for bad-request
+
+            if (!ModelState.IsValid || user.Id != 0 || nicknameExists == true || emailExists == true)
```

Comment:
Codestyle: "nicknameExists == true" should be replaced with "nicknameExists", ==true is alway ommitted.